bgpd/bgp_vty.c

```diff
@@ -16982,40 +17040,125 @@ static void bgp_config_write_peer_af(struct vty *vty, struct bgp *bgp,
 		}
 	} else {
 		if (peer->afc[afi][safi]) {
-			if ((afi == AFI_IP || afi == AFI_IP6)
-			    && safi == SAFI_UNICAST) {
-				if (afi == AFI_IP
-				    && CHECK_FLAG(bgp->flags,
-						  BGP_FLAG_NO_DEFAULT_IPV4)) {
+			if (afi == AFI_IP) {
```

Comment:
Wouldn't be easier with `bool default_af[AFI_MAX][SAFI_MAX];` here instead of default_af_flags? I mean by iterating like:

```
if (!bgp->default_af[afi][safi])
	vty_out(vty, "  neighbor %s activate\n",
				addr);
```
etc.

Reply:
ping @taspelund :)

Reply:
Sorry, got caught up with that pesky dayjob ;-)

For this my intention was mainly to try and match the existing logic + extend it for other AFIs.
I'm not opposed to using a 2D array, but my reason for not using it was simply because it didn't occur to me at the time.

If you think this is the proper way to get this done, lmk and I can rewrite it to use that instead.

Reply:
I still think this is a better way to handle that with much less code and less chance to hit the bug in the future :shrug: